Add `kola-junit` script to generate JUnit XML files

jlebon opened the pull request:
There's a lot of friction associated with debugging pipeline kola test
failures. One must scroll a bunch and sift through the output to see
which test(s) failed, and if more information is needed, download the
archived artifacts, extract it, and look for the journal and console
there.

Let's make this easier by outputting JUnit XML files. Jenkins can
natively display test results in that format so it's much easier to see
which tests ran, which ones failed, and their outputs (in the Blue Ocean
view, it's the "Tests" tab which we haven't been using so far).

For space efficiency and to reduce noise, we default to skipping over
passed tests.

jlebon commented:
You can see what it looks like in https://jenkins-coreos-ci.apps.ocp.fedoraproject.org/blue/organizations/jenkins/jlebon-tmp2/detail/jlebon-tmp2/12/tests

And screenshot for posterity for when that job will be wiped.

![image](https://github.com/coreos/coreos-assembler/assets/11934099/3e13ec14-24d6-4ce2-90b9-6971202da74e)

dustymabe commented:
It would be great in this new page view if we could see the stdout from `cosa kola run` that related to this test.. i.e. things like the test timed out, but I don't think that's easy to get at all :)

dustymabe commented on src/kola-junit:
```diff
@@ -0,0 +1,121 @@
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("classname")
+parser.add_argument("koladir")
+parser.add_argument("outputxml")
```
Looking at https://github.com/coreos/coreos-ci-lib/pull/156 I think it would be easier to follow if these were `--` parameters: `--classname`, `--koladir`,...

jlebon commented on mantle/cmd/kola/testiso.go:
```diff
@@ -491,6 +493,19 @@ func runTestIso(cmd *cobra.Command, args []string) error {
+	}
+
+	reporter := reporters.NewJSONReporter("report.json", "testiso", "")
+	defer func() {
+		if reportErr := reporter.Output(reportDir); reportErr != nil && err != nil {
+			err = reportErr
+		}
+	}()
```
Nice catch!

jlebon commented:
Updated for comments and added a freeze on osbuild as discussed in https://github.com/coreos/coreos-assembler/pull/3671! (Fine to break that out into its own PR if you'd prefer.)

> It would be great in this new page view if we could see the stdout from `cosa kola run` that related to this test.. i.e. things like the test timed out, but I don't think that's easy to get at all :)

That actually *is* possible. It was already the case for non-external tests, but now I've tweaked it so we retain the information for external tests too.

dustymabe commented on build.sh:
```diff
@@ -44,6 +44,9 @@ install_rpms() {
 
     frozendeps=""
 
+    # We freeze the version for now since we're carrying patches.
+    frozendeps+=" $(echo osbuild{,-ostree,-selinux,-tools}-100-1.fc39.noarch)"
+
```
I did something like this over in https://github.com/coreos/coreos-assembler/pull/3675 but since there was `python3-osbuild-100-1.fc39.noarch` I couldn't briefly find a good way to express that in this `echo osbuild` form. I guess maybe the `python3-osbuild` package is tightly bound to the others so it shouldn't matter?

dustymabe reviewed (approved):
LGTM
